### PR TITLE
En 2413 fix duplicate key on tokens

### PR DESCRIPTION
**Pull request by ampie**
I encountered an intermittent duplicate key exception on storing the OAuth Token. This only happens on MySQL and only on certain nodes.  The code in the DAO that inserts the token checks for the existience of the token just before it inserts it. If the token already exists, the method returns immediately. The implication of this exception is that this check does not detect the token yet, but it gets inserted exactly as the rest of the method is executed. As a fix, I repeated this check in the 'catch' clause if the exception involved a duplicate key contraint violation, and then swallow the exception if the token already exists. This got rid of the intermittent exception entirely,

### CODE REVIEW

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
All is well

File: engine/src/main/java/org/entando/entando/aps/system/services/oauth2/OAuth2TokenDAO.java [hunk 1]
```diff
@@ -190,6 +193,18 @@ public void storeAccessToken(OAuth2AccessToken accessToken, OAuth2Authentication
             stat.executeUpdate();
             conn.commit();
         } catch (Exception t) {
+            CharArrayWriter stackTraceWriter = new CharArrayWriter();
```
Comment (joewhite101) [inline engine/src/main/java/org/entando/entando/aps/system/services/oauth2/OAuth2TokenDAO.java]: I don't think we can merge this. Looks like a potential security problem. We'll have to find the underlying race condition.

**Comment by joewhite101** [comment]
Overcome by events.

**Pull request closed without merging**